composables/pagination.ts

```diff
@@ -0,0 +1,69 @@
+import { useResizeObserver } from '@vueuse/core';
+import type { PaginationRequestDTO, PaginationResponseDTO } from '~/types/dtos';
+import { parsePaginationQueryParam } from '~/utils/pagination';
+
+export const usePagination = <T extends PaginationResponseDTO>(tableRef: Ref<any>) => {
+  const router = useRouter();
+  const route = useRoute();
+
+  const pageCount = ref(1);
+  const pageSizes = [5, 10, 15, 20];
+
+  const parsedPageNumber = parsePaginationQueryParam(route.query.pageNumber as string, 1);
+  const parsedPageSize = parsePaginationQueryParam(route.query.pageSize as string, 10);
+
+  const query = ref<PaginationRequestDTO>({
```

Comment:
Shouldn't be reactive

Reply:
Has to be reactive, because we wouldn't be able to apply watchers in components that are using the query from the composable